litestar/_openapi/parameters.py

```diff
@@ -206,6 +208,24 @@ def create_parameter_for_handler(
     dependency_providers = route_handler.resolve_dependencies()
     layered_parameters = route_handler.resolve_layered_parameters()
 
+    # if there's a single parameter 'query', surface its fields
+    if "query" in handler_fields and len(handler_fields) == 1:
+        if pydantic.VERSION.startswith("1"):  # pragma: no cover
+            model_fields: dict[str, pydantic.fields.FieldInfo] = {k: model_field.field_info for k, model_field in handler_fields["query"].annotation.__fields__.items()}  # type: ignore
+        else:
+            model_fields = dict(handler_fields["query"].annotation.model_fields)
```

Comment:
It seems adding the `query` only works with `pydantic` models. I think this should be refactored to allow for passing in any of the models that are supported i.e. `msgpsec`, `pydantic`, `attrs`, `dataclasses`.